### ISSUE TITLE
False positive warning on Spy declared outside of before/after/it block

### ISSUE THREAD

**Issue by guoliang** [tlvince/eslint-plugin-jasmine#344]
**Describe the bug**

The following snippet gives me the following lint warning
> Spy declared outside of before/after/it block

```javascript
const setupMockComponents = () => {
		componentInstanceDatagrid = {
			id: "compId4",
			displayName: "Component four",
			typedef: "Datagrid",
			datasource: "some datasource",
			getId: jasmine.createSpy("getId").and.returnValue("compId4"),
			getDatabearer: jasmine.createSpy("getDatabearer").and.returnValue(datagridDatabearer),
			getTypedef: jasmine.createSpy("getTypedef").and.returnValue("Datagrid"),
			refreshComponent: () => {},
			getBoundarySetting: jasmine.createSpy("getBoundarySetting").and.returnValue({}),
			updateBindings: jasmine.createSpy("updateBindings"),
		};
	};
	beforeEach(() => {
		setupMockComponents();
	});
```

Should this be okay? or I shouldn't declare spies like so?

**Context**

- Node.js version: v14
- eslint-plugin-jasmine version: 4.1.3
- eslint version: 7.32.0
- Operating system: win10
